cmake/define-ia2-wrapper.cmake

```diff
@@ -28,30 +28,31 @@ function(define_ia2_wrapper)
             get_target_property(PKEY ${target}_wrapped PKEY)
             list(APPEND LD_ARGS_FILES ${CMAKE_CURRENT_BINARY_DIR}/${TEST_NAME}_call_gates_${PKEY}.ld)
         endif()
-
     endforeach()
-    set(COPY_TARGETS ${DEFINE_IA2_WRAPPER_ORIGINAL_TARGETS})
-    list(TRANSFORM COPY_TARGETS PREPEND "copy_files_")
+    set(ORIGINAL_SRCS ${ALL_SRCS})
+    list(TRANSFORM ORIGINAL_SRCS REPLACE ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
 
     set(CALL_GATE_SRC ${CMAKE_CURRENT_BINARY_DIR}/${TEST_NAME}_call_gates.c)
     set(CALL_GATE_HDR ${CMAKE_CURRENT_BINARY_DIR}/${TEST_NAME}_call_gates.h)
 
     add_custom_command(
         OUTPUT ${CALL_GATE_SRC} ${CALL_GATE_HDR} ${LD_ARGS_FILES}
+               ${ALL_SRCS}
         COMMAND ia2-rewriter
             --output-prefix=${CMAKE_CURRENT_BINARY_DIR}/${TEST_NAME}_call_gates
-            ${ALL_SRCS}
+            --root-directory=${CMAKE_CURRENT_SOURCE_DIR}
+            --output-directory=${CMAKE_CURRENT_BINARY_DIR}
+            -p=${CMAKE_BINARY_DIR}
```

Comment:
What does this do? Is there a long option version?

Reply:
It sets the build path which we need so the tool finds the compile commands JSON (otherwise it'll look in parent directories of the source file's directory and won't find the JSON in the build dir). I'll document this.